compaction/compaction.cc

```diff
@@ -631,17 +631,30 @@ class compaction {
     }
 private:
     // Default range sstable reader that will only return mutation that belongs to current shard.
-    virtual flat_mutation_reader_v2 make_sstable_reader() const = 0;
+    virtual flat_mutation_reader_v2 make_sstable_reader(schema_ptr s,
+                                                        reader_permit permit,
+                                                        const dht::partition_range& range,
+                                                        const query::partition_slice& slice,
+                                                        tracing::trace_state_ptr,
+                                                        streamed_mutation::forwarding fwd,
+                                                        mutation_reader::forwarding) const = 0;
```

Comment:
Can we keep the old interface (mostly because it's impossible to misuse) and instead add an interposer to skip?

Reply:
An interposer will add its own slowdown too. Not sure.

Reply:
I think we should remove parameters that will never be used:
```
                                                        const query::partition_slice& slice,
                                                        tracing::trace_state_ptr,
                                                        streamed_mutation::forwarding fwd,
```